virttest/qemu_vm.py

```diff
@@ -1591,6 +1591,20 @@ def sort_key(dev):
                 'aw-bits': params.get('iommu_aw_bits')}
             devices.insert(QDevice('intel-iommu', iommu_params))
 
+        # Add device virtio-iommu, it must be added before any virtio devices
+        if (params.get_boolean('virtio_iommu') and
+                devices.has_device('virtio-iommu-pci')):
+            iommu_params = {"pcie_direct_plug":
+                            params.get("virtio_iommu_direct_plug", "yes")}
```

Comment:
Is there any chance for it to be `no`? So that's why we have to introduce `pcie_direct_plug`?

Reply:
Currently, if insert it to `pcie-root-port`, the guest will break during boot, and the default behavior in libvirt is to insert it to `pcie.0`, this is the reason set it to `yes`, but will it support `pcie-root-port`? I am not sure, some people still discussing it.

Reply:
> @PaulYuuu I see, or can we exclude `virtio-iommu-pci` from `is_pcie_device` list?

Actually I cannot answer this question right now, because its bus is `pcie.0`, and it has some similar options like `iommu_platform`, `ats`, `aer`, so it's a PCIe device indeed. But if it can only insert to pcie.0 in the future, we can consider again, because at that time, it is a pcie.0 reserved device.
> And maybe just remove `pcie_direct_plug` as a param, as I don't see a real case that really need it?
I think we can keep it here, since we basically do not configure the parameters of virtio-iommu, it is more convenient to leave `pcie_direct_plug` alone I think. I have a concern that we move `pcie_direct_plug` to `virtio_iommu_extra_params` and set it by default, but users re-define `virtio_iommu_extra_params` again, then the case may fail, because the default behavior is inserted virtio device to pcie-root-port.

I am not sure I got your point?

Reply:
@PaulYuuu I don't figure out a better solution so far, let's have this fix and we can improve it in future if we can.
But I still need another approval, at least for basic testing, do you know whose test will be affected for this change?

Reply:
Yes, virtio-iommu is getting ready to support x86, but not yet finished. I would @JinLiul here, but this time I will just paste the sanity results of aarch64.